src/shared/utilities/promptSetting.ts

```diff
@@ -0,0 +1,37 @@
+/*!
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import * as vscode from 'vscode'
+import { getLogger } from '../logger'
+import { SettingsConfiguration } from '../settingsConfiguration'
+
+export type PROMPT_NAME =
```

Comment:
types are not named with all caps. You can check this via `git grep! -E 'type [A-Z][A-Z]+'`.

PROMPT_NAME is a questionable name here, because (1) we have an unrelated "Prompt" concept elsewhere in the codebase, and (2) this is a setting. This raises the question (from many previous PRs) of why we are naming this at all? It could be defined inline in the `disablePrompt()` signature.

Reply:
I'm guessing the rationale behind this was to have any callers add their own 'key' to the union type. But that seems rather cumbersome IMO. Consumers of the module should not need to change it to be able to use it. I suppose it'd be better to just use a similar style to `localize` where callers need to define their own key?

Reply:
I think it makes sense to have the valid keys defined in one place.

Reply:
All new types and interfaces removed, not needed here.